src/core/qgstemporalutils.cpp

```diff
@@ -51,15 +53,57 @@ QgsDateTimeRange QgsTemporalUtils::calculateTemporalRangeForProject( QgsProject
             layerRange = rasterLayer->dataProvider()->temporalCapabilities()->availableTemporalRange();
             break;
         }
+        break;
+      }
+
+      case QgsMapLayerType::VectorLayer:
+      {
+        QgsVectorLayer *vectorLayer = qobject_cast<QgsVectorLayer *>( currentLayer );
+
+        QgsVectorLayerTemporalProperties *properties = static_cast< QgsVectorLayerTemporalProperties * >( vectorLayer->temporalProperties() );
+        switch ( properties->mode() )
+        {
+          case QgsVectorLayerTemporalProperties::ModeFixedTemporalRange:
+            layerRange = properties->fixedTemporalRange();
+            break;
+
+          case QgsVectorLayerTemporalProperties::ModeFeatureDateTimeInstantFromField:
+          {
+            const int fieldIndex = vectorLayer->fields().lookupField( properties->startField() );
+            if ( fieldIndex >= 0 )
+            {
+              layerRange = QgsDateTimeRange( vectorLayer->minimumValue( fieldIndex ).toDateTime(),
+                                             vectorLayer->maximumValue( fieldIndex ).toDateTime() );
+            }
+            break;
+          }
+
+          case QgsVectorLayerTemporalProperties::ModeFeatureDateTimeStartAndEndFromFields:
```

Comment:
I'd make the start or end field optional here. The logic would be:
- if start and end fields given, filter is start_field >= current_datetime and end_field <= current_datetime
- if only start field given, filter is start_field >= current_datetime
- if only end field given, filter is end_field >= current_datetime

As for the QgsDateTimeRange values
- if start and end fields given, range is min(start_field) max(end_field)
- if only start field given, range is min(start_field) max(start_field)
- if only end field given, range is min(end_field) max(end_field)

Reply:
Done (but with the correct logic :wink:)